external-crates/move/crates/move-vm-runtime/src/execution/interpreter/eval.rs

```diff
@@ -161,9 +181,17 @@ fn step(
 
             profile_close_frame!(
                 gas_meter,
-                arena::to_ref(current_frame.function).pretty_string()
+                state.call_stack.current_frame.function().pretty_string()
```

Comment:
I think this is a bug in the current tracer as well, but I think above you'll need to do something like 
```rust
let err = gas_meter.charge_drop_frame(...).map_err(|e| ..);
profile_close_frame!( ...);
trace(..., Some(&err));
err?;
```

Otherwise if the charge for the drop frame causes an OOG you wont see it in the trace.